lib/gooddata/lcm/actions/synchronize_ldm.rb

```diff
@@ -83,23 +90,34 @@ def sync_segment_ldm(params, segment_info)
           segment_info[:from_blueprint] = blueprint
           maql_diff = nil
           previous_master = segment_info[:previous_master]
+          synchronize_ldm_mode = params[:synchronize_ldm].downcase
           diff_against_master = %w(diff_against_master_with_fallback diff_against_master)
-            .include?(params[:synchronize_ldm].downcase)
-          GoodData.logger.info "Synchronize LDM mode: '#{params[:synchronize_ldm].downcase}'"
-          if previous_master && diff_against_master
-            maql_diff_params = [:includeGrain]
-            maql_diff_params << :excludeFactRule if exclude_fact_rule
-            maql_diff_params << :includeDeprecated if include_deprecated
-            maql_diff = previous_master.maql_diff(blueprint: blueprint, params: maql_diff_params)
+            .include?(synchronize_ldm_mode)
+          GoodData.logger.info "Synchronize LDM mode: '#{synchronize_ldm_mode}'"
+          if segment_info.key?(:previous_master) && diff_against_master
```

Comment:
why don't we use previous_master instead of segment_info.key?(:previous_master) ?

Reply:
Because release brick use synchronize_ldm action to diff clients between dev workspace and new master workspace
So we need to skip release brick in this logic and go through to diff clients.
To detect running brick is release brick or rollout brick, we base on previous_master key in segment_info because it only exists in rollout_brick at collect_segment_clients action.